### DISCUSSION TITLE
Using `brew update` seems to break some application schemes

### DISCUSSION THREAD

**Discussion by qupig** [Homebrew/discussions#5429] [Getting started]
### Description of issue

When some apps are updated using `brew update someapp`, the related schemes of the app may be broken.

For example, it may not be possible by double-clicking to open a file with an associated extension (especially if the App is already open). Or open it through the context menu->open with->select the corresponding App. Or drag and drop files onto the App dock icon.

The issue seems to be that the update app has not been authenticated by Apple's pop-up window "Do you want to open this app downloaded from the Internet?" and re-register the relevant LaunchServices? (I'm not sure about the details)

Unless you open the App through these files before opening the App, it may trigger Apple to re-pop up the authorization window, but the problem is that this will destroy the Sessions history of some Apps.

An example is vscode, if I open vscode directly after brew update, it will show the session (all open states and projects) from the last time it was closed, but at that point I can't easily drag and drop a file onto the vscode dock icon to open that file. Unless I close the whole vscode and do the same, it brings up the Apple authorization pop-up and then the file is opened but the last sessions is lost.

One work around is to avoid using `brew update someapp` and instead use `brew uninstall theapp; brew install theapp` to update the app so that Apple's authorization pop-up even when the app is opened directly after the update and the last sessions are retained.

That is, The `brew update someapp` process seems to need some improvements to avoid this issue.

I should mention that the same situation exists even with `brew reinstall someapp`, I'm not sure why.

**Comment by rezzawi** [comment]
Not the most knowledgable when it comes to the use of terminal and coding in general, but I can say that ever since doing brew update a few days ago, apps have been closing randomly as you are saying. 

Is there a way to stop this? Maybe updating to the most recent OS?

I am currently on macOS Monterey version 12.7.2 - MacBook Pro (13-inch, 2019, Four Thunderbolt 3 Ports).

**Reply by rezzawi** [reply]
*Apps that close the most are Word (use of maximaWM, and gnuplot (gnuplot installed but doesn't work)) and safari. It has happened once with Unity and GitHub Desktop though.

**Reply by qupig** [reply]
I didn't say "closing randomly", perhaps an irrelevant issue.

**Comment by 301166** [comment]
I starting

**Reply by 301166** [reply]
I go and go forever end this case all!

**Comment by 301166** [comment]
> I starting
